### ISSUE TITLE
reclass.table masks the wrong road pixels

### ISSUE THREAD

**Issue by qdread** [kristalj/SESYNC-NEON-DATA#8]
I believe that the code used to mask out the roads is giving incorrect results. The code in question is:

```
reclass.table <- matrix(c(1,6,1,7,14,NA), ncol=3) #reclassify values 1-6 into 1 for keep drop the rest

imp.roads <- reclassify(imp.surf.mask, reclass.table)
```

You can see that imp.surf.mask has values between 1 and 11

```
table(values(imp.surf.mask))

     1      2      3      4      5      6      7     10     11 
829339   9702   1141  25851   8195 136085  20587 237970  24435 
```

So we would expect based on the comment that we'd have either 1 or NA in the reclassified raster. But it doesn't. In fact it still has all of those values except for 7.

```
table(values(imp.roads))

     1      2      3      4      5      6     10     11 
829339   9702   1141  25851   8195 136085 237970  24435 
```

I think this is because the reclass.table is not correct. We should add the argument byrow = TRUE when we create it:

```
reclass.table <- matrix(c(1,6,1,7,14,NA), ncol=3, byrow=TRUE)
```

Then we would end up with the correct reclassification that will keep only the road pixels (classes 1 through 6), turning them to 1, and mask the rest to NA.

Anyway, I can fix it and then test a bit to confirm it's fixed. At that point I think we will need to regenerate all the dasymetric rasters.

**Comment by qdread** [comment]
Also I think `right=NA` has to be added to `reclassify()` so that the intervals include the endpoints. So a complete fix would be 

```
reclass.table <- matrix(c(1,6,1,7,14,NA), ncol=3, byrow=TRUE)
imp.roads <- reclassify(imp.surf.mask, reclass.table, right = NA)
```

Another thing is that the `projectRaster()` causes weird behavior with the numeric class values so it might be good to convert to factor before projecting. This may not matter in this case since it only matters for non-NA pixels.

**Comment by qdread** [comment]
Potential bigger bug found!

The overlay function to mask roads by setting all pixels in the impervious ratio raster to NA where road is non-NA, actually sets all pixels to NA where road is NA. That's the opposite of what we want!!!

The code:

```
#Mask out roads (i.e, all NonNA values in imp.roads.p)
RISA <- overlay(lu.ratio.zp, imp.roads.p, fun = function(x, y) {
  x[is.na(y[])] <- NA
  return(x)
})
```

The fix should just be to put a `!` before the `is.na(y)`.

What RISA (downtown Annapolis) looks like before the fix is below. Looks like it is keeping roads which is bad.

![image](https://user-images.githubusercontent.com/9200842/125373832-cd3c3d80-e353-11eb-9d26-43223736eb3d.png)

What it looks like after the fix. Keeping non-roads which is good.

![image](https://user-images.githubusercontent.com/9200842/125373873-ddecb380-e353-11eb-98e0-0b732e8099e6.png)

So I will implement that fix too.

**Comment by qdread** [comment]
Final bug, this time it was something I did: I had been adding 1 to the impervious surface descriptor raster because I thought it was needed to convert from 0-based to 1-based indexing. Apparently that is not the case and it actually messed stuff up. So I got rid of that line too. All should now be fixed as of commit 6b9476b. I will close this issue once all code is rerun and we confirm it's correct.
